go/vt/vtgate/vtgate.go

```diff
@@ -583,6 +584,38 @@ func (vtg *VTGate) Execute(
 	return session, nil, err
 }
 
+// ExecuteMulti executes multiple non-streaming queries.
+func (vtg *VTGate) ExecuteMulti(
+	ctx context.Context,
+	mysqlCtx vtgateservice.MySQLConnection,
+	session *vtgatepb.Session,
+	sqlString string,
+) (newSession *vtgatepb.Session, qrs []*sqltypes.Result, err error) {
+	queries, err := vtg.executor.Environment().Parser().SplitStatementToPieces(sqlString)
+	if err != nil {
+		return session, nil, err
+	}
+	if len(queries) == 0 {
+		return session, nil, sqlparser.ErrEmpty
+	}
+	var qr *sqltypes.Result
+	var cancel context.CancelFunc
+	for _, query := range queries {
+		func() {
+			if mysqlQueryTimeout != 0 {
+				ctx, cancel = context.WithTimeout(ctx, mysqlQueryTimeout)
+				defer cancel()
+			}
```

Comment:
we should create an issue to deprecate and remove this, we already have `query-timeout`

Reply:
or we keep the meaning of this at mysql protocol level and not a per query timeout in which case it moves back to mysql package, what you think?

Reply:
Let's keep it as it is now.